docs:#1338 consolidate cluster config pages (#1603)
redpanda-data/documentation: modules/manage/pages/iceberg/specify-iceberg-schema.adoc

| The `format` keyword can be used for custom Iceberg types. See <<format-translation,`format` annotation translation>> for details.

| object 
| struct or map

> | Use `properties` to define `struct` fields and constrain their types. `additionalProperties: false` is supported for closed objects. A schema-valued `additionalProperties` object is translated to an Iceberg `map<string, T>`. You cannot combine `properties` and schema-valued `additionalProperties` in the same object.

Reviewer: @nvartolomei do I understand this correctly?

- if object contains `properties` (and optionally `additionalProperties: false`), -> struct
- if object contains `additionalProperties` whose value is itself a schema object -> map

Reviewer 2: Yes. That's it.

Reviewer: ```suggestion
a| * Use `properties` to define `struct` fields and constrain their types. `additionalProperties: false` is supported for closed objects. 
* If `additionalProperties` contains a schema, it translates to an Iceberg `map<string, T>`.
* You cannot combine `properties` and `additionalProperties`in an object if `additionalProperties` is set to a schema.
```
I'd suggest a tweak to the formatting in this cell so it's a little easier to scan and to see the different conditions in which the type is translated to struct vs map. Please feel free to adjust the wording for accuracy

|===

Keyword behavior:

> * The `$ref` keyword is supported for internal references resolved from schema resources declared in the same document (using `$id`), including relative and absolute URI forms. References to external resources and references to unknown keywords are not supported. A root-level `$ref` schema is not supported.

Reviewer 3: ```suggestion
* The `$ref` keyword is supported for internal references resolved from schema resources declared in the same document (using `$id`), including relative and absolute URI forms. References to external resources and references to undefined schema resources are not supported. A root-level `$ref` schema is not supported.
```

Reviewer 2: unknown keywords is the right term
i.e.
> A JSON Schema MAY contain properties which are not schema keywords. Unknown keywords SHOULD be ignored.
> https://json-schema.org/draft-07/draft-handrews-json-schema-01
